src/include/indexlr-workers.h

```diff
@@ -338,16 +332,22 @@ MinimizeWorker::work()
 
 			auto minimizers = getMinimizers(hashes, w);
 
-			if (stlfr) {
-				std::regex right("^.*#+");
-				std::regex left("/.*");
-
-				read.barcode = std::regex_replace(read.id, right, "");
-				read.barcode = std::regex_replace(read.barcode, left, "");
+			if (read.id.find("BX:Z:") == std::string::npos) {
```

Comment:
```suggestion
			if (read.barcode.empty()) {
```

Reply:
`read.barcode` is not empty. It will contain `NA`.

Reply:
so I could do `read.id.find("BX:Z:") == std::string::npos` or `read.barcode=="NA"`

Reply:
```suggestion
			if (read.barcode == "NA") {
```